src/meta/meta_data.h

```diff
@@ -324,6 +324,11 @@ class app_state_helper
             cc.lb_actions.clear();
         }
     }
+
+    void reset_manual_compact_status();
+    // get replica group manual compact progress
+    // return false if partition is not executing manual compaction
+    bool get_manual_compact_progress(/*out*/ int32_t &progress) const;
```

Comment:
What about implement like this: 
```int32_t get_manual_compact_progress() const;```
The return value is progress. 0 means is not executing manual compaction

Reply:
The return value is to distinguish not executing manual compaction and compaction is running or queuing but progress is 0.

Reply:
But I can't get these two cases in your implementation of `get_manual_compact_progress`

Reply:
I got your point. There are other manual compaction status: queuing and running, both of them is not finished status.